daemon/graphdriver/btrfs/btrfs.go

```diff
@@ -288,13 +307,22 @@ func subvolDelete(dirpath, name string) error {
 	return nil
 }
 
-func (d *Driver) subvolEnableQuota() error {
-	if d.quotaEnabled {
-		return nil
-	}
-	// In case quotaEnabled is not set, check qgroup and update quotaEnabled as needed
-	if _, err := subvolLookupQgroup(d.home); err == nil {
+func (d *Driver) isQuotaEnabled() bool {
+	if !d.quotaEnabled {
+		// In case quotaEnabled is not set, check qgroup and update quotaEnabled as needed
+		if err := subvolQgroupStatus(d.home); err != nil {
+			// quota is still not enabled
+			return false
+		}
 		d.quotaEnabled = true
+	}
+	return true
+}
+
+func (d *Driver) subvolEnableQuota() error {
+	d.mu.Lock()
```

Comment:
Thanks @mlaventure.

Inside the `subvolEnableQuota` the `d.quotaEnabled` will be changed again to `true` at the end of `subvolEnableQuota` (located in Line 343). For that I placed the `Lock()` outside of `isQuotaEnabled()`.